pkg/broker/ingress/multi_topic_decouple_sink_test.go

```diff
@@ -220,6 +226,224 @@ func TestMultiTopicDecoupleSink(t *testing.T) {
 	}
 }
 
+func TestHasTrigger(t *testing.T) {
+	DecoupleQueue := &config.Queue{Topic: "test_topic", State: config.State_READY}
+
+	tests := []struct {
+		name          string
+		brokerTargets map[string]*config.Target
+		hasTrigger    bool
+	}{
+		{
+			name:       "broker with no target",
+			hasTrigger: false,
+		},
+		{
+			name:          "broker with empty target",
+			brokerTargets: map[string]*config.Target{},
+			hasTrigger:    false,
+		},
+		{
+			name: "broker with target with no filters",
+			brokerTargets: map[string]*config.Target{
+				"target_1": {},
+			},
+			hasTrigger: true,
+		},
+		{
+			name: "broker with target with matching filter",
+			brokerTargets: map[string]*config.Target{
+				"target_1": {
+					FilterAttributes: map[string]string{
+						"type":   eventType,
+						"source": "test-source",
+					},
+				},
+			},
+			hasTrigger: true,
+		},
+		{
+			name: "broker with target with non-matching filter",
+			brokerTargets: map[string]*config.Target{
+				"target_1": {
+					FilterAttributes: map[string]string{
+						"type":   eventType,
+						"source": "some-random-source",
+					},
+				},
+			},
+			hasTrigger: false,
+		},
+		{
+			name: "broker with multiple targets with one matching target filter",
```

Comment:
Can you add a test of multiple targets with none matching?

Reply:
Sure. done.